src/widget/wcuemenupopup.cpp

```diff
@@ -135,6 +187,65 @@ void WCueMenuPopup::slotDeleteCue() {
     hide();
 }
 
+void WCueMenuPopup::slotSavedLoopCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pBeatLoopSize.valid()) {
+        return;
+    }
+    if (m_pCue->getType() == mixxx::CueType::Loop) {
+        m_pCue->setType(mixxx::CueType::HotCue);
+    } else {
+        m_pCue->setType(mixxx::CueType::Loop);
+        auto cueStartEnd = m_pCue->getStartAndEndPosition();
+        if (!cueStartEnd.endPosition.isValid() ||
+                cueStartEnd.endPosition <= cueStartEnd.startPosition) {
+            double beatloopSize = m_pBeatLoopSize.get();
+            const mixxx::BeatsPointer pBeats = m_pTrack->getBeats();
+            if (beatloopSize <= 0 || !pBeats) {
+                return;
+            }
+            m_pCue->setEndPosition(pBeats->findNBeatsFromPosition(
+                    cueStartEnd.startPosition, beatloopSize));
+        }
+    }
+    slotUpdate();
+}
+
+void WCueMenuPopup::slotAdjustSavedLoopCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pBeatLoopSize.valid()) {
+        return;
+    }
+    const mixxx::BeatsPointer pBeats = m_pTrack->getBeats();
+    auto position = mixxx::audio::FramePos::fromEngineSamplePos(
+            m_pPlayPos.get() * m_pTrackSample.get());
+    m_pCue->setType(mixxx::CueType::Loop);
+    if (!m_pQuantizeEnabled.toBool() || !pBeats) {
+        m_pCue->setEndPosition(position);
```

Comment:
We need to reject if we're at the cue position, otherwise we'll hit
https://github.com/mixxxdj/mixxx/blob/4164341f5063b58fd9251e75620d48ed9ad18a6d/src/engine/controls/loopingcontrol.cpp#L655 when activating the cue.